### DISCUSSION TITLE
Rewrite does not work

### DISCUSSION THREAD

**Discussion by chiqui3d** [vercel/next.js#19382] [Help]
# Bug report

## Describe the bug

I'm trying to reuse the pages with `rewrite `and all I get are 404 errors

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

Example:

1. Config

```
const config = {
    i18n: {
        locales: ['en','es','de','no','se'],
        defaultLocale: 'en',
        localeDetection: false,
    },
    async rewrites() {
        return [
				  { source: '/', destination: '/' },
				  { source: '/es/', destination: '/' },
				  { source: '/de/', destination: '/' },
				  { source: '/no/', destination: '/' },
				  { source: '/se/', destination: '/' },
				  { source: '/es/blog/', destination: '/es/blog' },
				  { source: '/es/politica-privacidad/', destination: '/privacy' },
				  { source: '/privacy-policy/', destination: '/privacy' },
				  {
				    source: '/de/datenschutzerklarung-rechtlicher-hinweis/',
				    destination: '/privacy'
				  },
				  { source: '/no/personvern-juridisk-notis/', destination: '/privacy' },
				  {
				    source: '/se/integritetspolicy-rattsligt-meddelande/',
				    destination: '/privacy'
				  },
				  { source: '/es/politica-cookies/', destination: '/cookies-policy' },
				  { source: '/cookies-policy/', destination: '/cookies-policy' },
				  {
				    source: '/de/richtlinie-fur-cookies/',
				    destination: '/cookies-policy'
				  },
				  { source: '/no/cookiespolicy/', destination: '/cookies-policy' },
				  {
				    source: '/se/cookies-kakor-policy/',
				    destination: '/cookies-policy'
				  }
				]
    },
```

2. When I navigate to `/es/politica-cookies/`, I only get a 404 error, I just try to reuse the "cookies-policy.js" page, for all languages. How complicated is it?

I have also tried to reulize the same page with `getStaticPaths `and I couldn't either

3. 

## Expected behavior

The expected behavior is that a 404 will not return, and be able to reuse my pages either for languages or something else.

## System information

- OS: Windows 10 
- Browser: Chrome
- Version of Next.js: 10.0.2
- Version of Node.js: 12.16.1
- Deployment: next dev

## Additional context

`!important`: There is no possibility that something like this exists https://nuxtjs.org/docs/2.x/features/file-system-routing#extending-the-router, and not be so complex?

**Comment by chiqui3d** [comment]
I also have to add, that I don't understand why I can't add query parameters in `Rewrites`, I find the route system a bit rare.

I refer to have the option of having an alternative routing system, ie something similar to `Rewrite` but fully functional and has the options (query, redirect, etc ...), this will solve the amount of methods that are being applied to avoid this, making a more stable system for more complex pages, not for simple pages.

**Comment by ijjk** [comment]
Hi, it looks like you need to be using `locale: false` for these rewrites as mentioned in [the docs here](https://nextjs.org/docs/api-reference/next.config.js/rewrites#rewrites-with-i18n-support)

**Comment by chiqui3d** [comment]
Same effect: Error 404

Thanks!

**Comment by chiqui3d** [comment]
It seems that with the Canary version if it is working, I will do some more tests tomorrow and I already confirm.

Thanks

**Comment by chiqui3d** [comment]
The routes I have in a subdirectory, under a specific locale, are not working, since I only want the blog in **one language** for now, so based on the same routes of the first post and adding `locale: false`, it doesn't work. The other routes that are not related to the blog are working.

`{ source: '/es/blog/', destination: '/es/blog', locale: false },`

This route should handle the destination of: `/es/blog/index.js`, but it does not. 

I have also tried to remove the locale on that route, but it doesn't work either.

I have to do it differently? to have the blog in only one language.

**Comment by chiqui3d** [comment]
I found a temporary solution, but it didn't help, because when I was exporting my app, I got a message that **I18n is not compatible with export**. Anyway, I put it here just in case.


Instead of putting the blog in the directory `/es/blog/index.js`, I have put it in `/blog/index.js`, and all the requests that are not the language of the blog I response with `404`.

```js
export const getStaticProps = async (context) => {
    const language = context.locale;
    if (language !== 'es'){
        return {
            notFound: true,
        }
    }
    const posts = await getLastPosts(10);
    const {default: lngDict = {}} = await import(`../../../public/locales/${language}/common.json`);
    return {
        props: {
            posts,
            language,
            lngDict
        }
    }

}
```

I would also like this to be solved, as it is ignoring the previous route with `locale: false`, also being able to have static pages with i18n compatibility

> All these final tests have been done with the canary version

**Comment by chiqui3d** [comment]
I just hit another wall.

Because it seems that I18n doesn't support `export` :(, I decided to redo all my application without `rewrite `and make the paths with exportPathMap, in `exportPathMap `it doesn't work either because I can't pass neither locale nor `query parameters` to be obtained with `getStaticProps`.


Any solution?


> I have seen that with getInitialProps , I can get what I want with exportPathMap , but I will never get a completely static page, which is what I am looking for.
